qiskit/algorithms/evolvers/classical_methods/scipy_real_evolver.py

```diff
@@ -0,0 +1,328 @@
+        """
+        op_mean = np.array([np.real(state.conjugate().dot(op.dot(state))) for op in aux_ops])
+        op_std = np.sqrt(
+            np.array([np.real(state.conjugate().dot(op2.dot(state))) for op2 in aux_ops_2])
+            - op_mean**2
+        )
+        return op_mean, op_std
+
+    def _start(
+        self, evolution_problem: EvolutionProblem
+    ) -> Tuple[
+        np.ndarray,
+        sp.csr_matrix,
+        sp.csr_matrix,
+        List[sp.csr_matrix],
+        List[sp.csr_matrix],
+        int,
+        float,
+    ]:
+        """Returns a tuple with the initial state as an array, the operators needed for time
+         evolution as sparse matrices, the number of timesteps in which to divide the time evolution
+         and the tolerance for the BiCG method.
+
+        Args:
+            evolution_problem: The definition of the evolution problem.
+
+        Returns:
+            A tuple with the initial state as an array, the operators needed for time
+            evolution as sparse matrices, the operators to evaluate at each timestep as a list
+            of sparse matrices as well as a list of the squared operators to compute the standard
+            deviation, the number of timesteps in which to divide the time evolution and
+            the tolerance for the BiCG method.
+        """
+        # Convert the initial state and hamiltonian into sparse matrices.
+        if isinstance(evolution_problem.initial_state, QuantumCircuit):
+            state = Statevector(evolution_problem.initial_state).data.T
+        else:
+            state = evolution_problem.initial_state.to_matrix(massive=True).transpose()
+
+        hamiltonian = evolution_problem.hamiltonian.to_spmatrix()
+
+        # Determine the number of timesteps.
+        timesteps = min(
+            self._ntimesteps(
+                time=evolution_problem.time, hamiltonian=hamiltonian, threshold=self.threshold
+            ),
+            self.max_iterations,
+        )
+        timestep = evolution_problem.time / timesteps
+
+        # Create the operators for the time evolution.
+        idnty = sp.identity(hamiltonian.shape[0], format="csr")
+
+        lhs_operator = idnty + 1j * timestep / 2 * hamiltonian
+        rhs_operator = idnty - 1j * timestep / 2 * hamiltonian
+
+        if isinstance(evolution_problem.aux_operators, list):
+            aux_ops = [aux_op.to_spmatrix() for aux_op in evolution_problem.aux_operators]
+        elif isinstance(evolution_problem.aux_operators, dict):
+            aux_ops = [aux_op.to_spmatrix() for aux_op in evolution_problem.aux_operators.values()]
+        else:
+            aux_ops = []
+
+        aux_ops_2 = [aux_op.dot(aux_op) for aux_op in aux_ops]
+
+        return (
+            state,
+            lhs_operator,
+            rhs_operator,
+            aux_ops,
+            aux_ops_2,
+            timesteps,
+            self._bicg_tol(timesteps),
```

Comment:
This function seems a little unneccessary, maybe just a local variable would be enough? 🙂